### ISSUE TITLE
Using FZF turns off `relativenumber`

### ISSUE THREAD

**Issue by raichoo** [neovim/neovim#6204]
- `nvim --version`: v0.2.0-747-g286aacb2
- Vim (version: ) behaves differently? N/A
- Operating system/version: Archlinue
- Terminal name/version: gnome-terminal
- `$TERM`:  gnome-256color
### Actual behaviour

Using FZF turns of the `relativenumber` setting in the buffer I'm currently working in.

### Expected behaviour

That should not happen.

### Steps to reproduce using the following configuration

```
call plug#begin('~/.nvim/plugged')

Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
Plug 'junegunn/fzf.vim'

call plug#end()

set number
set relativenumber
```
- load a file
- issue `:Files` command
- exit FZF window using `ESC`
- relative lines are now turned off

Reverting 857113ca8c76cd103b752ad4460ee1b091e0831a helps.

**Comment by justinmk** [comment]
Thanks for reporting.

**Comment by gkapfham** [comment]
Hello @justinmk! Thanks for your hard work on `nvim`. I know that you have closed this issue. However, I am still seeing this problem with the following version of `nvim` on Ubuntu 16.04: `NVIM v0.2.1-dev`. My impression is that the fix should already be working in the version that I am running. Any updates that you could provide would be very much appreciated. Thanks again!

**Comment by justinmk** [comment]
Did you exit fzf?

**Comment by gkapfham** [comment]
Hello @justinmk! Thanks for looking into this issue.

I can confirm that if I:

- Start `nvim` with `set relativenumber` as enabled by default
- Confirm that `relativenumber` is working correctly
- Open `fzf` with a command like `:FZF`
- And then either exit by pressing `ESC` or by selecting a file

I will now see that `relativenumber` is no longer working for that buffer.

Please let me know if you need any more information about this issue. Thanks again for your work on `nvim` and your investigation of this matter.

**Comment by justinmk** [comment]
see https://github.com/neovim/neovim/pull/6796, now we don't modify terminal buffer-local 'number' nor 'relativenumber' at all. However 'relativenumber' is still ignored in _terminal mode_ (i.e. "insert mode, in a terminal buffer") because it's just completely broken. But it doesn't affect the option value.

**Comment by gkapfham** [comment]
Hello @justinmk! I wanted to write to thank you as this issue is now fixed when I am running `NVIM v0.2.1-dev` with the most recent updates on Ubuntu 16.04 LTS. Again, thanks for your work on this project --- I regularly enjoy using NeoVim on a wide variety of different projects.
